accepted/future-releases/records/records-feature-specification.md

```diff
@@ -586,8 +586,51 @@ The runtime type of `pair` is `(int, double)`, not `(num, Object)`, However, the
 variable declaration is still valid and sound because records are naturally
 covariant in their field types.
 
+### Interactions with libraries using older language versions
+
+The records feature is language versioned, as usual with new Dart features.
+This means that it will be an error to use the syntax for records in libraries
+which do not have a language version greater than or equal to the language
+version in which records are released.  More specifically, assuming that `v` is
+the language version in which records are released, the following errors apply.
+
+It is an error to reference the SDK class `Record` in a library whose language
+version is less than `v`.
+
+It is an error for the record literal syntax (e.g. `(3, 4)`) to be used
+syntactically in a library whose language version is less than `v`.
+
+It is an error for the record type syntax (e.g. `(int, int)`) to be used
+syntactically in a library whose language version is less than `v`.
+
+*It is not an error for a library whose language version is less than `v` (a
+"legacy library") to include or reference the `Record` class, record types or
+record expressions directly or indirectly from another library whose language
+version is greater than or equal to `v`.  Such a legacy library may reference a
+typedef name which is bound to a record type in another library, and the
+semantic interpretation of the typedef is as the underlying record type, just as
+it would be for any other type.  Similarly, type inference may introduce record
+types into a legacy library, and such types will be interpreted by the compiler
+as record types as usual (that is, there is no erasure implied to remove these
+inferred types).  Record values may flow into a legacy library via a reference
+to a member from another libraries, and a legacy library may freely call getters
+on record values (since there is no new syntax for calling a record getter).
+The reason for this choice is that the intent of language versioning (for an
```

Comment:
The "this choice" is a little underspecified (there are many small choices above, and normative and explanatory text is not distinguished).

Just elaborate: "The choice to not make it an error to reference record types in legacy libraries ...."

(Documentation always gets better when the implicit is made explicit. Prove me wrong!)

Reply:
More verbosity isn't always better?  This text is commentary - the set of errors is completely specified and circumscribed without it.  Do you really find this text confusing?  If we're trying to be 100% formally precise it's going to be a lot more wordy.  For example, it's not just "The choice to not make it an error to reference record types".  It's "The choice to make X, Y and Z and error, but not A, B, C, D, E, F is because" (because note that it's not just types, it's expressions, etc, etc).  On balance, unless you legitimately believe that this non-normative text is in some way confusing, I'd prefer to err on the side of legibility here.

Reply:
Sharpened "this choice" slightly, PTAL.

Reply:
Verbosity isn't better, but being explicit is better than being implicit :)

Phrasing is better now. I don't get confused about which choice "this choice" refers to.

Excellent :)